src/core/typing.ts

```diff
@@ -46,21 +43,24 @@ export interface Typing extends EmitsDiscontinuities {
    * Get the current typers, a set of clientIds.
    * @returns A Promise of a set of clientIds that are currently typing.
    */
-  get(): Promise<Set<string>>;
+  get(): Set<string>;
 
   /**
-   * Start indicates that the current user is typing. This will emit a typingStarted event to inform listening clients and begin a timer,
-   * once the timer expires, a typingStopped event will be emitted. The timeout is configurable through the typingTimeoutMs parameter.
-   * If the current user is already typing, it will reset the timer and being counting down again without emitting a new event.
+   * Start indicates that the current user is typing.
+   * This will emit a `typing.start` event to inform listening clients and begin a heartbeat timer,
+   * which can be configured through the `heartbeatIntervalMs` parameter.
+   * If the current user is already typing, this will no-op until the heartbeat timer has elapsed,
+   * at which point calling `start()` will restart the timer and emit a new `typing.start` heartbeat.
    *
    * @returns A promise which resolves upon success of the operation and rejects with an ErrorInfo object upon its failure.
    */
 
   start(): Promise<void>;
 
   /**
-   * Stop indicates that the current user has stopped typing. This will emit a typingStopped event to inform listening clients,
-   * and immediately clear the typing timeout timer.
+   * Stop indicates that the current user has stopped typing.
+   * This will emit a `typing.stop` event to inform listening clients,
+   * and immediately clear any active timers.
```

Comment:
```suggestion
   * Stop indicates that the current user has stopped typing.
   * This will emit a `typing.stopped` event to inform listening clients.
```